Sources/GDSCommon/UI/UIKit/SharedViews/Patterns/IconScreenView/IconScreenViewController.swift

```diff
@@ -0,0 +1,48 @@
+import UIKit
+
+public final class IconScreenViewController: UIViewController {
+    public override var nibName: String? { "IconScreenView" }
+    
+    let viewModel: IconScreenViewModel
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public init(viewModel: IconScreenViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: "IconScreenView", bundle: .module)
+    }
```

Comment:
Something that I missed on a couple of screens that I need to go back and fix was setting the back button and the right bar button. Whilst most of the time they might both be `nil` it adds a level of flexibility, especially if this screen is ever used in modal presentation style. Happy to have a chat about it tomorrow.

Reply:
Not sure tbh. Adding the properties to the view model later will change the public API which is better to avoid if we can avoid it. But yes, it is going to happen elsewhere.